### PR TITLE
Research director should be able to buy science-related items in uplink

### PR DESCRIPTION
**Pull request by Kistras**
## About the PR
Research director is now able to buy science-related items in uplink in case they turn out to be traitor (possibly via adminverb).
There is also a `BuyerDepartmentCondition` so that adding department specific items is possible.

**Media**

You already know that it exists though

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:

### CODE REVIEW

**Comment by lzk228** [comment]
then make it with the other heads too

**Comment by metalgearsloth** [comment]
what they said

**Comment by Kistras** [comment]
> then make it with the other heads too

There isn't anyone else besides research director who were left out, but I added HoP and Captain to the list of other items. I guess...
HoP doesn't get access to science related items because they aren't research director.

Tell me if there is anything else I should add.

**Comment by lzk228** [comment]
no, i mean other department things to department heads

**Comment by lzk228** [comment]
and hop with cap dont really need to have all that things

**Comment by Kistras** [comment]
> no, i mean other department things to department heads

No such thing exists

**Comment by Kistras** [comment]
> no, i mean other department things to department heads

No such thing exists. Only two departaments that get job-specific items are:
Civilian: gatfruit (botanist), gloves (passenger/boxer), Necronomicon (chaplain), holy grenade (chaplain), cap gun (mime/clown), explosive banana peel (clown), hot potato (chef/botanist/clown/mime), explosive wet floor sign (janitor)
Science: omega CHIMP ammo (scientist, senior researcher)

Also there are fake monkey cubes (zookeeper, scientist, senior researcher, chef) which kinda belong to two departaments.
This is comprehensive list. I don't get what I should do.

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Heads are already strong enough as traitors, but even with that out of the way the proper way to do this is by adding a `BuyerDepartmentCondition` instead of adding to a list of jobs

**Comment by Kistras** [comment]
> I think heads having command-only traitor items would make more sense, but even with that out of the way the proper way to do this is by adding a `BuyerDepartmentCondition` instead of adding to a list of jobs

Done that. It's not possible to use it at SyndicateSpongeBox though with current system as chef or zookeeper can also get it.

**Comment by Kistras** [comment]
Forgot to push oof

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 1]
```diff
@@ -964,10 +964,12 @@
...
   conditions:
-    - !type:BuyerJobCondition
+    - !type:BuyerDepartmentCondition
       whitelist:
-        - Scientist
-        - SeniorResearcher
+      - Science
+    - !type:BuyerJobCondition
+      blacklist:
+      - ResearchAssistant # No fun for trainee roles
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:972]: just remove this. this seems like an oversight anyway

**Review by EmoGarbage404 — APPROVED** [review]
ack

File: Resources/Prototypes/Catalog/uplink_catalog.yml (hunk 1 above)
Comment (Kistras) [reply Resources/Prototypes/Catalog/uplink_catalog.yml]: Removed it. Should assistants also get SyndicateSpongeBox?

**Pull request merged**